### PR TITLE
Adding update to fix_audit_syscall_rule to not use slashes

### PR DESCRIPTION
**Pull request by MollyJoBault**
Some of the audit rules (eg content_rule_audit_rules_privileged_commands_userhelper) have filenames that need to be inserted.  

`-a always,exit -F path=/usr/bin/userhelper -F perm=x -F auid>=1000 -F auid!=4294967295 -F key=privileged`

This is fine and dandy but when it gets put into sed the slashes are not escaped and they get choked on.  

```
sed: -e expression #1, char 26: unknown command: `u'
sed: -e expression #1, char 26: unknown command: `u'
sed: -e expression #1, char 26: unknown command: `u'
```

A simple fix is just to change sed's separator to an underscore.

### CODE REVIEW

**Comment by mpreisler** [comment]
Hmm, ok but if the filenames have an underscore this is still a problem. Underscore is pretty common in filenames.

**Comment by redhatrises** [comment]
What about using semicolons as sed's separator?

**Comment by redhatrises** [comment]
Please undo commit e68c808 and look at changing the separator to a semicolon.

File: shared/bash_remediation_functions/fix_audit_syscall_rule.sh [hunk 1]
```diff
@@ -111,7 +113,8 @@ do
 	# * follow the rule pattern, and
 	# * meet the hardware architecture requirement, and
 	# * are current syscall group specific
-	IFS=$'\n' existing_rules=($(sed -e "/${pattern}/!d" -e "/${arch}/!d" -e "/${group}/!d"  "$audit_file"))
+	IFS=$'\n' existing_rules=($(sed -e ";${pattern};!d" -e "/${arch}/!d" -e "/${group}/!d"  "$audit_file"))
+	retval=$?
```
Comment (mpreisler) [inline shared/bash_remediation_functions/fix_audit_syscall_rule.sh]: ?

Comment (MollyJoBault) [inline shared/bash_remediation_functions/fix_audit_syscall_rule.sh]: Sorry, you are going to have to give me more than a question mark. :)

Comment (mpreisler) [inline shared/bash_remediation_functions/fix_audit_syscall_rule.sh]: why is it here but not on line 87, what is this about (nothing in commit message, no comments)

Comment (MollyJoBault) [inline shared/bash_remediation_functions/fix_audit_syscall_rule.sh]: I updated and make the "retval" logic more sticky.

**Comment by dahaic** [comment]
Hello @MollyJoBault the changes look good. If you can squash this PR into one commit, and fix the indentation to be all tabs, it's OK for merge.

**Comment by MollyJoBault** [comment]
Sorry about the tabs, I didn't see that vim wasn't a valid editing tool for this file.

**Comment by dahaic** [comment]
Sorry for not noticing earlier - can you revert the executable flag on the file?

**Comment by MollyJoBault** [comment]
And this is why I've been doing all my edits in vim (which this is not tab compliant to.  Over samba it makes them executable.  I'm running non-graphical.  Anyway, the fix is made.

**Comment by dahaic** [comment]
Great, thanks!

(Sidenote - even `vim` can tab tabs :) )

**Pull request merged**